sklearn/compose/_column_transformer.py

```diff
@@ -522,7 +525,10 @@ def _get_column(X, key):
     if column_names:
         if hasattr(X, 'loc'):
             # pandas dataframes
-            return X.loc[:, key]
+            if not callable(key):
```

Comment:
Can't we do this before the ifs, setting key = key(X)?

Reply:
Yes, will do